File: 7_ReverseInteger/reverseInteger.java
public class reverseInteger {
    class Solution {
        public static int reverse(int x) {
            long pow = 1;
            for(int i=0; i<31; i++){
                pow = pow * 2;
            }
            long ans = 0;
    
            while (x!=0){
                ans *= 10;
                ans += x%10;
                x = x/10;
            }
    
            if(ans>=pow || ans <=-1 * pow){
                return 0;
            }
    
            return (int) ans;
        }
    }

    public static void main(String[] args){
        int x = 123;

        System.out.println(Solution.reverse(x));
    }
}

File: 7_ReverseInteger/ReverseInteger.cpp
#include <iostream>

using namespace std;

class Solution {
public:
    static int reverse(int x) {
        long long int ans=0;
        
        while (x!=0){
            ans *= 10;
            ans += x%10;
            x = x/10;
        }

        if(ans>=2147483648 || ans <=-2147483648){
            return 0;
        }

        return ans;
    }
};

int main(){

    int x = 1534236469;

    cout << Solution::reverse(x) << endl;

    return 0;
}